python/cudf/cudf/core/column/column.py

```diff
@@ -1512,6 +1512,7 @@ def build_column(
             children=children,
         )
     elif is_struct_dtype(dtype):
+        assert size is not None
```

Comment:
We shouldn't rely on assertions, how can we be more explicit?

Reply:
Replaced the assertions with TypeErrors, add TODOs to make the `size` param optional in column constructors.